### PR TITLE
Remove undefined value from records to prevent attachment loading

### LINKED ISSUE #4247: Record Set Attachment Viewer skeleton loader with and without attachments
**Describe the bug**

Browse in Forms attachment viewer icon shows a skeleton viewer when multiple forms in browse. expected behavior still happening when a single record is in view (see recording)

**Expected behavior**

Should pull up "There are no attachments" dialog.

**Screenshots**

Screenshot 2023-12-01 at 3 20 42 PM
Screenshot 2023-12-01 at 3 21 47 PM

https://github.com/specify/specify7/assets/135047322/7f1c3511-574c-49ad-81ab-401b4942783f

**Desktop:**
 - OS: macOS Sonoma
 - Browser: Chrome
 - Specify 7 Version: v7.9.2

**Database Name:** `aafc_aac_3_8_23`

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4247

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/Attachments/RecordSetAttachment.tsx [hunk 1]
```diff
@@ -46,9 +46,9 @@ export function RecordSetAttachments<SCHEMA extends AnySchema>({
         )
       );
 
-      const fetchCount = records.findIndex(
-        (record) => record?.populated !== true
-      );
+      const fetchCount = records
+        .filter((record) => record !== undefined)
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/Attachments/RecordSetAttachment.tsx:50]: use filterArray

File: specifyweb/frontend/js_src/lib/components/Attachments/RecordSetAttachment.tsx [hunk 2]
```diff
@@ -46,9 +46,9 @@ export function RecordSetAttachments<SCHEMA extends AnySchema>({
...
+        .findIndex((record) => record?.populated !== true);
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/Attachments/RecordSetAttachment.tsx:51]: don't need `?` anymore here

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]
see https://github.com/specify/specify7/pull/4252/files#r1414916180

**Review by maxpatiiuk — APPROVED** [review]

**Review by bronwyncombs — APPROVED** [review]
Looking good! No more skeleton loading in attachment viewer :-)

Record Set with attachments: 
Screenshot 2023-12-06 at 9 17 54 AM
Record Set without attachments:
Screenshot 2023-12-06 at 9 18 29 AM

Browse in Forms records w/ attachments:
Screenshot 2023-12-06 at 9 26 23 AM

Browse in Forms records w/o attachments:

Screenshot 2023-12-06 at 9 27 46 AM

https://github.com/specify/specify7/assets/135047322/3ab4eadd-92d4-4f3c-b934-ed0d10a9c686

**Pull request merged**